modules/apps/analytics/analytics-layout-page-template-web/src/main/java/com/liferay/analytics/layout/page/template/web/internal/servlet/taglib/AnalyticsRenderFragmentLayoutPreDynamicInclude.java

```diff
@@ -90,13 +95,27 @@ private void _printAnalyticsCloudAssetTracker(
 			return;
 		}
 
+		if (Objects.equals("file", analyticsAssetType)) {
+			FileEntry fileEntry = (FileEntry)displayObject;
```

Comment:
Could be an option to call the `sendAnalyticsEvent` JS method from here?

Reply:
We can't use the data-analytics-file-entry-id approach, because the metrics are sent directly yo AC in Documents and Media using the approach: 

https://github.com/liferay/liferay-portal/blob/3f82556b44ec230b8f3fd159db470829d80e8611/modules/apps/document-library/document-library-analytics/src/main/resources/META-INF/resources/dynamic_include/view_file_entry.jsp.

So we have analyze again the document.js code of analytics cloud https://github.com/liferay/liferay-portal/blob/3f82556b44ec230b8f3fd159db470829d80e8611/modules/apps/analytics/analytics-client-js/src/plugins/documents.js and just adding the attribute `analytics-asset-action="preview" ` everything should work fine.

Thanks @beltranrengifo and @claraizquierdo for your contribution.